gr-fec/lib/async_decoder_impl.cc

```diff
@@ -112,88 +107,66 @@ void async_decoder_impl::decode_unpacked(pmt::pmt_t msg)
 
     size_t o0(0);
     const float* f32in = pmt::f32vector_elements(bits, o0);
-    pmt::pmt_t outvec(pmt::make_u8vector(nbits_out, 0x00));
+    const size_t nitems_out = d_packed ? nbits_out / 8 : nbits_out;
+    pmt::pmt_t outvec(pmt::make_u8vector(nitems_out, 0x00));
     uint8_t* u8out = pmt::u8vector_writable_elements(outvec, o0);
+    uint8_t* decoder_out = d_packed ? d_bits_out.data() : u8out;
+    const float shift = d_decoder->get_shift();
 
     if (strncmp(d_decoder->get_input_conversion(), "uchar", 5) == 0) {
```

Comment:
I also didn't like this `strncmp` when I saw it. Besides the `"ucharacterless" == "uchar"`, I found more annoying that it's not very readable.

I think it makes sense to change this now, instead of opening a new PR just to change this. I have replaced this by:
`std::string_view(d_decoder->get_input_conversion()) == "uchar"`
It seems to work as intended (note that `get_input_conversion()` returns a `char *`).